src/include/common/exception.h

```diff
@@ -129,6 +129,8 @@ class Exception : public std::runtime_error {
     }
   }
 
+  ExceptionType GetType() { return type; }
```

Comment:
With the changes to your test, do you still need this?

Reply:
Yes, because we need to ensure that the exception of the appropriate type was thrown.

Reply:
Is this being used anymore? If not remove it.